js/apps/admin-ui/src/realm-settings/RealmSettingsTabs.tsx

```diff
@@ -199,14 +199,15 @@ export const RealmSettingsTabs = ({
     }
 
     try {
-      await adminClient.realms.update(
-        { realm: realmName },
-        {
-          ...realm,
-          ...r,
-          id: r.realm,
-        }
-      );
+      const savedRealm = {
+        ...realm,
+        ...r,
+        id: r.realm,
+      };
+
+      // For default value, back end is expecting null instead of empty string
+      if (savedRealm.smtpServer?.port === "") savedRealm.smtpServer.port = null;
```

Comment:
Note that even though a new object is created before, it still contains a reference to the original realm since `realm.smtpServer` is nested and spreading an object using the spread operator (`...`) only creates a shallow clone.

If we want to accomplish this without mutating the original reference we can do the following:
```suggestion
      // For the default value, null is expected instead of an empty string.
      if (savedRealm.smtpServer?.port === "") {
        savedRealm.smtpServer = { ...savedRealm.smtpServer, port: null };
      }

```

Reply:
@jonkoops thank you for pointing out.